sdktests/server_side_data_store_updates.go

```diff
@@ -0,0 +1,225 @@
+package sdktests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/launchdarkly/go-test-helpers/v2/jsonhelpers"
+	m "github.com/launchdarkly/go-test-helpers/v2/matchers"
+	"github.com/launchdarkly/sdk-test-harness/framework/ldtest"
+	"github.com/launchdarkly/sdk-test-harness/mockld"
+	"github.com/launchdarkly/sdk-test-harness/servicedef"
+
+	"gopkg.in/launchdarkly/go-sdk-common.v2/lduser"
+	"gopkg.in/launchdarkly/go-sdk-common.v2/ldvalue"
+	"gopkg.in/launchdarkly/go-server-sdk-evaluation.v1/ldbuilders"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func doServerSideDataStoreStreamUpdateTests(t *ldtest.T) {
+	// These tests verify that the SDK's default data store correctly implements updates and
+	// deletes with versioning.
+	//
+	// Since our mechanism for putting updates into the SDK is our stream test fixture, these
+	// tests also verify that the stream is working correctly.
+
```

Comment:
We have unit tests equivalent to this in most/all of the SDKs, but they're tedious to write and it's easy to miss edge cases.

Integration-harness can only verify the basic "an update is persisted if it has a higher version" case, because integration-harness has no way of causing LD to emit out-of-order updates.